runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/FlinkStreamingTransformTranslators.java

```diff
@@ -284,8 +286,16 @@ public void flatMap(T s, Collector<WindowedValue<T>> collector) throws Exception
               }
             }).assignTimestampsAndWatermarks(new IngestionTimeExtractor<WindowedValue<T>>());
       } else {
-        source = context.getExecutionEnvironment()
-            .addSource(new UnboundedSourceWrapper<>(context.getPipelineOptions(), transform));
+        try {
+          UnboundedSourceWrapper<T, UnboundedSource.CheckpointMark> sourceWrapper =
+              new UnboundedSourceWrapper<>(
+                  context.getPipelineOptions(),
+                  (UnboundedSource) transform.getSource(),
```

Comment:
Nope, it was there because `transform.getSource()` returns a `UnboundedSource<T, ?>` while I wanted to have a `UnboundedSource<T, UnboundedSource.CheckpointMark>`. For some reason the CheckpointMark generic parameter is `?` in almost all of the code. I'll change it to `?` here as well to avoid the `SuppressWarnings`.